### PR TITLE
Stats: Deprecate feature flag "stats/new-main-chart" within shared components

### PR DESCRIPTION
**Pull request by dognose24**
#### Proposed Changes

* Introduce prop isLegacy to component `StatsNavigation`.
* Refactor component `StatsDatePicker` with the short title.
* Refactor component `StatsPeriodNavigation` with the new structure.
* Refactor component `Delta` with new suffix styles.
* Refactor modernized `StatsNavigation` styles.
* Refactor modernized `FormattedHeader` styles.
* Remove the modernized page header styles file.

#### Testing Instructions

> Note: The instructions to show the Store and Ads pages can be found at https://github.com/Automattic/wp-calypso/pull/69661 and https://github.com/Automattic/wp-calypso/pull/69743.

* Spin up this change with the Calypso Live link.
* Navigate to the Stats `Traffic`, `Insights`, `Store`, `Ads`, `Overview (/stats/day)`, and `Google My Business (/google-my-business/stats/${your-site})` pages.
* Ensure the styles are the same as the previous ones.
* a12s: Since the `Google My Business` tab would be hard to reproduce, we can `su` into a local instance with details from `p1670518452086579-slack-C0438NHCLSY`.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #70209

### CODE REVIEW

File: client/my-sites/google-my-business/stats/index.js
Comment (jsnmoon) [inline client/my-sites/google-my-business/stats/index.js:209]: How can I test this?

**Review by jsnmoon — CHANGES_REQUESTED** [review]
The changes look reasonable, but this PR needs testing instructions for the Google My Business tab of Calypso Stats.

File: client/my-sites/google-my-business/stats/index.js
Comment (dognose24) [reply client/my-sites/google-my-business/stats/index.js:209]: @jsnmoon Thanks for the information `p1670518452086579-slack-C0438NHCLSY` about making the `Google My Business` tab appear, which I have updated to the test instructions. I have tested this PR on the `Google My Business` page, and it works as expected. 🙇🏼

截圖 2022-12-09 下午4 50 59

**Review by jsnmoon — CHANGES_REQUESTED** [review]
Looks good! I want to do one more review after merge conflicts have been resolved, given the large size of this PR.

**Comment by dognose24** [comment]
> Looks good! I want to do one more review after merge conflicts have been resolved, given the large size of this PR.

Of course! The [trunk merging](https://github.com/Automattic/wp-calypso/pull/70564/commits/c899d64d0333b9541aa787cede2e71ffcbe0cfb5) has been done.

**Review by jsnmoon — CHANGES_REQUESTED** [review]
This change appears to add non-functional period navigation buttons to [the Stats Overview pages](https://wordpress.com/stats/day).

|Before|After|
|-|-|
|image|image|

Is this caused by this PR? If so, could we prevent the period buttons from rendering on the overview pages?

**Comment by dognose24** [comment]
> This change appears to add non-functional period navigation buttons to [the Stats Overview pages](https://wordpress.com/stats/day).

The period navigation buttons appeared there intentionally. I think it was missing due to the [feature flag](https://github.com/Automattic/wp-calypso/pull/70564/files#diff-c48db36d0f71bd2871975d28a9aaeca7d92c3bcfc04e227e1eddee46109dd285L58).

> Is this caused by this PR? If so, could we prevent the period buttons from rendering on the overview pages?

No. The period navigation buttons were broken because of the [routes I tried to fix in another PR](https://github.com/Automattic/wp-calypso/pull/70613).

Since #70613 is merged into the trunk, I believe everything in this PR will be fine after merging the trunk.

**Review by jsnmoon — APPROVED** [review]
Looks great. Thanks for your patience!

**Pull request merged**
